src/test/scala/org/graphframes/PregelSuite.scala

```diff
@@ -58,14 +58,14 @@ class PregelSuite extends SparkFunSuite with GraphFrameTestSparkContext {
     val pageRankResultDF = new Pregel(graph1)
       .setMaxIter(15)
       .withVertexColumn("rank", lit(1.0 / N),
-        when(Pregel.msg.isNotNull, Pregel.msg).otherwise(col("rank")))
+        coalesce(Pregel.msg, lit(1.0 - pr_alpha) / N))
       .sendMsgToDst(col("src.rank") / col("src.outDegree"))
       .aggMsgs(sum(col(Pregel.MSG_COL_NAME)) * lit(pr_alpha) + lit(1.0 - pr_alpha) / N)
       .run()
 
     val res = pageRankResultDF.sort(col("id"))
       .select("rank").as[Double].collect()
-    res.zip(Array(0.194, 0.195, 0.252, 0.194, 0.2)).foreach { case (a, b) =>
+    res.zip(Array(0.095, 0.111, 0.133, 0.095, 0.030)).foreach { case (a, b) =>
```

Comment:
The new result seems wrong to me because the ranks should sum up to 1.0. Could you double check and add a test on the sum?